src/calc.y

```diff
@@ -22,12 +22,34 @@ Factor -> Result<i128, ()>:
 %%
 // Any functions here are in scope for all the grammar actions above.
 
-fn parse_int(s: &str) -> Result<i128, ()> {
+pub enum ErrorType {
```

Comment:
Is `ErrorType` a good name? Given that there are many error types in Rust and maybe also in your own apps.

Maybe something like `InterpError` would be more appropriate?

Reply:
changed